src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -57,43 +129,96 @@ def __repr__(self):
             prefix = 'Raw Device'
         return '<%s: %s>' % (prefix, self.abspath)
 
+    def pretty_report(self):
+        output = ['\n====== Device report {} ======\n'.format(self.path)]
+        output.extend(
```

Comment:
the last thing I see could be improved is special casing the rejected reasons. Right now displays a Python list. If the list grows to more than one item it defeats the purpose of using the `pretty` option (might as well just parse the JSON).

When a device is not available it should be pretty clear that it isn't (to avoid having the user scrolling to look for it). Some items that might help would be making the title/header red, and placing the `rejected` always first (or always last?).

When the rejected reasons are displayed, they shouldn't use the list, they should be rendered one per line, something like:

    ====== Device report /dev/vdg ======

         rejected reasons          
                 * locked

Reply:
Sorting is already implemented. Devices are sorted by the valid flag (valid < invalid) and the device name otherwise.

Reply:
RIght, but if you are displaying 100 devices and you have 84 devices that are invalid, you still need to browse through quite a bunch to find where the division between valid/invalid starts.

Again, I am not saying this is necessary for this PR, if prioritizing, I think the rejected reasons fix would be preferable

Reply:
I could flip the ordering. Maybe it makes more sense though to add the ability to simply not display invalid devices?

Reply:
From my opinion the default should be listing everything.
We have 3 categories of drives ,
- totally free to be used as new OSDs
- used by ceph
- used by anything else

It could be useful to have flags to keep only a single category to ease the reading for humans or make an orchestrator having a way to pre-filter the devices types like "please tell me what devices are free to be used".

Reply:
I think this would mostly be interesting for the pretty report. Any software piece that interacts with the json output can easily filter this output and an orchestrator implementation might actually want that info (how many unavailable disks).

I do think this would be best addressed in a separate PR.